packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -154,10 +154,15 @@ export class AssetsContractController extends BaseControllerV1<
       this.configure({ ipfsGateway });
     });
 
-    onNetworkDidChange((networkState) => {
-      if (this.config.chainId !== networkState.providerConfig.chainId) {
+    onNetworkDidChange(({ selectedNetworkClientId }) => {
+      const selectedNetworkClient = getNetworkClientById(
+        selectedNetworkClientId,
+      );
+      const { chainId } = selectedNetworkClient.configuration;
+
+      if (this.config.chainId !== chainId) {
```

Comment:
Is there a test-case for when `chainId` differs? If not, that could explain the reduced coverage %?

Reply:
I looked into this and I don't think it's due to additional logic. Somehow I have made something that was inadvertently tested not tested anymore. Regrettably I have added an `istanbul` directive to ignore the logic in question: https://github.com/MetaMask/core/pull/4254/commits/be7f9c2b55b8ff0ef12dc9004196cc48b74e3dbc

Reply:
This change disappeared when this PR was broken up into multiple PRs.